src/webhook/validation/proxy_url_test.go

```diff
@@ -0,0 +1,164 @@
+						Value:     validEncodedProxyUrl,
+						ValueFrom: "",
+					},
+				},
+			})
+	})
+
+	t.Run(`invalid proxy url`, func(t *testing.T) {
+		assertDeniedResponse(t,
+			[]string{errorInvalidActiveGateProxyUrl},
+			&dynatracev1beta1.DynaKube{
+				ObjectMeta: defaultDynakubeObjectMeta,
+				Spec: dynatracev1beta1.DynaKubeSpec{
+					APIURL: testApiUrl,
+					Proxy: &dynatracev1beta1.DynaKubeProxy{
+						Value:     invalidPlainTextProxyUrl,
+						ValueFrom: "",
+					},
+				},
+			})
+	})
+
+	t.Run(`valid proxy secret url`, func(t *testing.T) {
+		assertAllowedResponseWithoutWarnings(t,
+			&dynatracev1beta1.DynaKube{
+				ObjectMeta: defaultDynakubeObjectMeta,
+				Spec: dynatracev1beta1.DynaKubeSpec{
+					APIURL: testApiUrl,
+					Proxy: &dynatracev1beta1.DynaKubeProxy{
+						Value:     "",
+						ValueFrom: testProxySecret,
+					},
+				},
+			},
+			&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testProxySecret,
+					Namespace: testNamespace,
+				},
+				Data: map[string][]byte{
+					"proxy": []byte(validEncodedProxyUrl),
+				},
+			})
+	})
+
+	t.Run(`missing proxy secret`, func(t *testing.T) {
+		assertDeniedResponse(t,
+			[]string{errorMissingActiveGateProxySecret},
+			&dynatracev1beta1.DynaKube{
+				ObjectMeta: defaultDynakubeObjectMeta,
+				Spec: dynatracev1beta1.DynaKubeSpec{
+					APIURL: testApiUrl,
+					Proxy: &dynatracev1beta1.DynaKubeProxy{
+						Value:     "",
+						ValueFrom: testProxySecret,
+					},
+				},
+			})
+	})
+
+	t.Run(`invalid format of proxy secret`, func(t *testing.T) {
+		assertDeniedResponse(t,
+			[]string{errorInvalidProxySecretFormat},
+			&dynatracev1beta1.DynaKube{
+				ObjectMeta: defaultDynakubeObjectMeta,
+				Spec: dynatracev1beta1.DynaKubeSpec{
+					APIURL: testApiUrl,
+					Proxy: &dynatracev1beta1.DynaKubeProxy{
+						Value:     "",
+						ValueFrom: testProxySecret,
+					},
+				},
+			},
+			&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testProxySecret,
+					Namespace: testNamespace,
+				},
+				Data: map[string][]byte{
+					"invalid-name": []byte(validEncodedProxyUrl),
+				},
+			})
+	})
+
+	t.Run(`invalid proxy secret url`, func(t *testing.T) {
+		assertDeniedResponse(t,
+			[]string{errorInvalidProxySecretUrl},
+			&dynatracev1beta1.DynaKube{
+				ObjectMeta: defaultDynakubeObjectMeta,
+				Spec: dynatracev1beta1.DynaKubeSpec{
+					APIURL: testApiUrl,
+					Proxy: &dynatracev1beta1.DynaKubeProxy{
+						Value:     "",
+						ValueFrom: testProxySecret,
+					},
+				},
+			},
+			&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testProxySecret,
+					Namespace: testNamespace,
+				},
+				Data: map[string][]byte{
+					"proxy": []byte(invalidPlainTextProxyUrl),
+				},
+			})
+	})
+
+	t.Run(`invalid proxy secret url - entrypoint.sh`, func(t *testing.T) {
+		assert.Equal(t, true, isStringValidForAG("password"))
+		assert.Equal(t, true, isStringValidForAG("test~!@#^*()_-|}{[]\":;?><./pass"))
```

Comment:
sorry I didn't see that during the first review. There is `assert.True` and `assert.False` for these kinds of assertions. Otherweise lgmt

Reply:
good point, fixed